### ISSUE TITLE
Pocketbook - random crashes

### ISSUE THREAD

**Issue by darthrevan** [koreader/koreader#10663]
* KOReader version: 2023.06 Hydrangea
* Device: Pocketbook 614 (Basic 2) (I know that it's an old device but I really like the physical page turning buttons and koreader is so much faster than the built-in software, if only I could get it to work)

https://github.com/koreader/koreader/blob/master/frontend/device/pocketbook/device.lua#L457

#### Issue
KOReader keeps crashing randomly with "illegal instruction" error.
 
#### Steps to reproduce
Just open the app and start reading a book. It will crash within a minute or two when a button is pressed. I could reproduce this while flipping the pages or while using the menus. I could reproduce the issue with different buttons. 

I understand that my device is ancient. I have tried a few older builds but the result was the same. If you could recommend an older build please let me know - I don't need any bells and whistles.

[crash.log](https://github.com/koreader/koreader/files/11992245/crash.log)

Thank you in advance!

**Comment by NiLuJe** [comment]
The only meaningful test would be 2022.01 (i.e., right before #8727), but I doubt it'll change much; usually, having loaded InkView is enough to make stuff crash; and we currently *have* to load InkView, at the very least for device identification, but for input, too (except on rooted devices).

**Comment by darthrevan** [comment]
I have tried 2022.01 and it has changed a lot. Thank you so much for the recommendation. 

Now it's only crashing if I'm using the menus while reading a book or if I'm in the files browser, but I can change the settings by editing the config files and if I setup the file type associations I'll be able to use the built-in files browser. Flipping through the pages seems to work well.

I understand that an older build won't get fixed but I can still post my crash logs if anyone is interested.

Thank you!

**Comment by NiLuJe** [comment]
If it's a segfault or a SIGILL (and it'll likely be that), that's (most likely) entirely not actionable on that platform anyway.

**Comment by darthrevan** [comment]
It's good to know 👍

**Comment by rjd22** [comment]
@NiLuJe it might be an idea to start deprecating older devices to actually allow less breakage on them. 

Older versions of KoReader are still really capable but we can't confirm if it's still working for older devices unless someone tells us so.

So would it be an idea to start removing devices that are for example a certain time without any updates from pocketbook to lock them onto a version that was still confirmed working?

**Comment by poire-z** [comment]
We're old, we love old books, many of use have old devices (because there's really nothing much new and they do their job) - so let's not play youth and go at deprecating anything that's old ! :)
We did for Android because of really problematic build toolchain issues.
But if there's no such hard reason with Pocketbook, we may as well keep old devices definitions and let them be detected and known - and have us or the user (ie. with user patches) deal with any issue - or live with some feature not working.

(ie. I have a Kobo Glo HD with the firmware still from 2015, and KSM, the launcher at the time - and I'm happy @NiLuJe has not yet killed support for it :)

**Comment by rjd22** [comment]
While I agree we should not deprecate devices just for the sake of it supporting older versions of devices does become problematic for newer devices from time to time.

I didn't see the deprecation of android while writing the previous message my message was based on the fact that the differences between the older and newer pocketbook devices keep becoming bigger. An example would be the difference of wifi states in older and newer devices.

**Comment by Frenzie** [comment]
> While I agree we should not deprecate devices just for the sake of it, supporting older versions of devices does become problematic for newer devices from time to time.

Yes, we can certainly deprecate older devices iff they become problematic. You're probably in a better position to judge that for PocketBook than most of us. But we do prefer to err on the side of not deprecating. ;-) Like @poire-z, I use an H2O from 2014 with firmware that hasn't been updated in probably years, though newer than the original firmware in order to fix it for some newer firmware that had broken. It also still receives updates from Kobo of course, or would if I installed any.

**Comment by NiLuJe** [comment]
For a platform like that, I could only really push for deprecation if you actually really needed to implement something code-wise that would be annoying to deal with for older devices.

Given that we murdered the legacy "DIY" codepath altogether for PB, and everybody runs on InkView, that's... not really a problem so far?

That stuff doesn't actually *work* is not really a problem, we never really had a PB maintaine,r so stuff has never really been all that fine, so it comes with a certain expectation that stuff may be janky.
